tests/services/pids/providers/test_datacite_pid_provider.py

```diff
@@ -72,11 +87,13 @@ def test_datacite_provider_reserve(record, datacite_provider):
     assert db_pid.status == PIDStatus.RESERVED
 
 
-def test_datacite_provider_register(record, datacite_provider, mocker):
+def test_datacite_provider_register(record_w_links, datacite_provider, mocker):
     mocker.patch("invenio_rdm_records.services.pids.providers.datacite." +
                  "DataCite43JSONSerializer")
-    created_pid = datacite_provider.create(record)
-    assert datacite_provider.register(pid=created_pid, record=record, url=None)
+    created_pid = datacite_provider.create(record_w_links._record)
```

Comment:
sanity check: in the code the flow goes as follows
1. the component launches the `register_pid` task passing the recid.
2. the task gets the record using `service.read`, this should return a `RecordItem` object.
3. this `RecordItem` is sent to the provider.

Shouldn't we need to pass just `record_w_links` here? because the `service.publish` of the fixture should return also ` RecordItem`. We should not need the `service.read` no? (line 48 above)

Could it be that the provider expects different object in the `create` and `publish` method? maybe we need to harmonize this.